src/app/qgsmaptoolcircle2points.cpp

```diff
@@ -34,7 +34,8 @@ void QgsMapToolCircle2Points::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
 
   if ( e->button() == Qt::LeftButton )
   {
-    mPoints.append( point );
+    if ( mPoints.size() < 1 )
```

Comment:
```suggestion
    if ( mPoints.empty() )
```

And doesn't this make the condition on the next if redundant?

Reply:
Right. Done.